File: model/Affectation.java
package com.chams.tpe_backend.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "affectation")
public class Affectation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "tpe_id", nullable = false)
    private Tpe tpe;

    @ManyToOne
    @JoinColumn(name = "station_id", nullable = false)
    private Station station;

    @Column(nullable = false)
    private LocalDateTime dateDebut = LocalDateTime.now();

    private LocalDateTime dateFin;

    // --- Getters et Setters ---

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Tpe getTpe() {
        return tpe;
    }

    public void setTpe(Tpe tpe) {
        this.tpe = tpe;
    }

    public Station getStation() {
        return station;
    }

    public void setStation(Station station) {
        this.station = station;
    }

    public LocalDateTime getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(LocalDateTime dateDebut) {
        this.dateDebut = dateDebut;
    }

    public LocalDateTime getDateFin() {
        return dateFin;
    }

    public void setDateFin(LocalDateTime dateFin) {
        this.dateFin = dateFin;
    }
}

File: flutter/lib/models/affectation.dart
import 'tpe.dart';
import 'station.dart';

class Affectation {
  final int id;
  final Tpe tpe;
  final Station station;
  final String dateDebut;
  final String? dateFin;

  Affectation({
    required this.id,
    required this.tpe,
    required this.station,
    required this.dateDebut,
    this.dateFin,
  });

  factory Affectation.fromJson(Map<String, dynamic> json) {
    return Affectation(
      id: json['id'],
      tpe: Tpe.fromJson(json['tpe']),
      station: Station.fromJson(json['station']),
      dateDebut: json['dateDebut'] ?? '',
      dateFin: json['dateFin'],
    );
  }
}